File: examples/symmetry/src/dns/client/dns_client.c
#include <stddef.h>

typedef struct { const char *resolver; unsigned long query_count; } dns_client_t;

void dns_client_start(dns_client_t *c, const char *resolver) { c->resolver = resolver; c->query_count = 0; }
const char *dns_client_lookup(dns_client_t *c, const char *name) { c->query_count++; return name; }

File: examples/symmetry/src/dns/client/DnsClient.java
package symmetry.dns.client;

public final class DnsClient {
    private String resolver;
    private long queryCount;

    public void start(String resolver) { this.resolver = resolver; queryCount = 0; }
    public String lookup(String name) { queryCount++; return name; }
    public String getResolver() { return resolver; }
    public long getQueryCount() { return queryCount; }
}
